File: python_solutions/1614.py
class Solution:
    def maxDepth(self, s: str) -> int:
        curr_depth ,max_depth = 0 , 0
        for c in s:
            if c == '(':
                curr_depth +=1 
            elif c == ')':
                curr_depth -= 1
            max_depth = max(max_depth,curr_depth)
        return max_depth
        

File: c++_solutions/1614.cpp
class Solution {
public:
    int maxDepth(string s) {
        int curr_depth = 0 , max_depth = 0 ; 
        for (char const & c  : s){
            switch(c){
                case '(':
                    curr_depth++;
                    break ;
                case ')':
                    curr_depth--;
                    break;
            }
            max_depth = max(max_depth , curr_depth);
        }
        return max_depth ;
    }
};
